packages/x-charts-pro/src/ChartZoomSlider/internals/ChartAxisZoomSlider.tsx

```diff
@@ -141,6 +147,136 @@ export function ChartAxisZoomSlider({ axisDirection, axisId }: ChartZoomSliderPr
+    pointerDownZoom = Math.max(Math.min(pointerDownZoom, maxEnd), minStart);
+
+    let pointerMoved = false;
+
+    const onPointerMove = rafThrottle(function onPointerMove(pointerMoveEvent: PointerEvent) {
+      const pointerMovePoint = getSVGPoint(element, pointerMoveEvent);
+      const pointerZoom = calculateZoomFromPoint(store.getSnapshot(), axisId, pointerMovePoint);
+
+      if (pointerZoom === null) {
+        return;
+      }
+
+      pointerMoved = true;
+      const zoomOptions = selectorChartAxisZoomOptionsLookup(store.getSnapshot(), axisId);
+
+      instance.setAxisZoomData(axisId, (prevZoomData) => {
+        if (pointerZoom > pointerDownZoom) {
+          const end = calculateZoomEnd(
+            pointerZoom,
+            { ...prevZoomData, start: pointerDownZoom },
+            zoomOptions,
+          );
+
+          /* If the starting point is too close to the end that minSpan wouldn't be respected, we need to update the
+           * start point. */
+          const start = calculateZoomStart(
+            pointerDownZoom,
+            { ...prevZoomData, start: pointerDownZoom, end },
+            zoomOptions,
+          );
+
+          return { ...prevZoomData, start, end };
+        }
+
+        const start = calculateZoomStart(
+          pointerZoom,
+          { ...prevZoomData, end: pointerDownZoom },
+          zoomOptions,
+        );
+
+        /* If the starting point is too close to the start that minSpan wouldn't be respected, we need to update the
+         * start point. */
+        const end = calculateZoomEnd(
+          pointerDownZoom,
+          { ...prevZoomData, start, end: pointerDownZoom },
+          zoomOptions,
+        );
+
+        return { ...prevZoomData, start, end };
+      });
+    });
+
+    const onPointerUp = function onPointerUp(pointerUpEvent: PointerEvent) {
+      rect.releasePointerCapture(pointerUpEvent.pointerId);
+      rect.removeEventListener('pointermove', onPointerMove);
+      document.removeEventListener('pointerup', onPointerUp);
+
+      if (pointerMoved) {
+        return;
+      }
+
+      // If the pointer didn't move, we still need to respect the zoom constraints (minSpan, etc.)
+      // In that case, we assume the start to be the pointerZoom and calculate the end.
+      const pointerUpPoint = getSVGPoint(element, pointerUpEvent);
+      const pointerUpZoom = calculateZoomFromPoint(store.getSnapshot(), axisId, pointerUpPoint);
+
+      if (pointerUpZoom === null) {
+        return;
+      }
+
+      const zoomOptions = selectorChartAxisZoomOptionsLookup(store.getSnapshot(), axisId);
+
+      instance.setAxisZoomData(axisId, (prev) => ({
+        ...prev,
+        start: pointerUpZoom,
+        end: calculateZoomEnd(pointerUpZoom, prev, zoomOptions),
+      }));
+    };
+
+    event.preventDefault();
+    event.stopPropagation();
+
+    rect.setPointerCapture(event.pointerId);
+    document.addEventListener('pointerup', onPointerUp);
```

Comment:
Couldn't make it work when `pointerup` listener was set on `rect`. The event wasn't being triggered in some cases.

Reply:
this is normal